packages/firestore/src/api/database.ts

```diff
@@ -1451,11 +1480,12 @@ export class DocumentSnapshot<T = firestore.DocumentData>
       this._key.isEqual(other._key) &&
       (this._document === null
         ? other._document === null
-        : this._document.isEqual(other._document))
+        : this._document.isEqual(other._document)) &&
+      isEqualConverter(this._converter, other._converter)
     );
   }
 
-  private convertObject(
+  private convertObjectValue(
```

Comment:
It would be nice to avoid overloading "convert"...  and try to choose different terminology for converting a FieldValue to plain JS vs converting plain JS to the user's custom model type.

Maybe:
convertValue => toJSValue()
convertObjectValue => toJSObject()
convertArray => toJSArray()

?

Reply:
Wow, this is much nicer!